src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -51,46 +52,121 @@ internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4__Item4 };
+
+        private static WellKnownMember[] tuple5Members = { WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5__Item1,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5__Item2,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5__Item3,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5__Item4,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5__Item5 };
+
+        private static WellKnownMember[] tuple6Members = { WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item1,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item2,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item3,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item4,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item5,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6__Item6 };
+
+        private static WellKnownMember[] tuple7Members = { WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item1,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item2,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item3,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item4,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item5,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item5,
+                                                            WellKnownMember.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7__Item7 };
+
+        private static WellKnownMember[][] tupleMembers = { tuple1Members, tuple2Members, tuple3Members, tuple4Members, tuple5Members, tuple6Members, tuple7Members };
```

Comment:
Consider inlining the `tupleNMembers` here and removing the extra declarations above.